### PR TITLE
Re-allow multiple modifiers per function.

### LINKED ISSUE #2288: Size and stack pos mismatch for multiple non-empty 'post condition' modifier
- Which version of Solidity you are using
0.4.11+commit.68ef5810.Emscripten.clang
- What was the source code (if applicable)
https://gist.github.com/be21f66e6cf208ce78cbd2dece7e1db9
- Which platform are you running on
https://ethereum.github.io/browser-solidity/#version=soljson-v0.4.11+commit.68ef5810.js&optimize=true
- How to reproduce the issue
See gist https://gist.github.com/be21f66e6cf208ce78cbd2dece7e1db9
- What was the result of the issue
`Internal compiler error (/src/libsolidity/codegen/LValue.cpp:50): Size and stack pos mismatch.
`
- What the expected behaviour is
No compilation error

### LINKED ISSUE #2621: False compiler DeclarationError: Modifier already used for this function
Solidity compiler throws an error 

`DeclarationError: Modifier already used for this function.`

when a function uses modifier twice with different parameters, e.g. 

```
modifier nonZeroAddress(address x) {
    require(x != 0);
    _;
  }

function X(address a, address b) 
nonZeroAddress(a)
nonZeroAddress(b)
{ ... }

```

### PR DESCRIPTION
**Pull request by chriseth**
Properly fixes https://github.com/ethereum/solidity/issues/2288
Fixes https://github.com/ethereum/solidity/issues/2621

### CODE REVIEW

**Comment by axic** [comment]
Tests fail

**Comment by chriseth** [comment]
Tests should be fixed now.

File: Changelog.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Features:
 Bugfixes:
  * Type Checker: Fix invalid "specify storage keyword" warning for reference members of structs.
  * Type Checker: Mark modifiers as internal.
+ * Type Checker: Re-allow multiple mentions of the same modifier per function.
```
Comment (axic) [inline Changelog.md:17]: `mentions`? `uses` is more fitting I think? Though this is not important.

File: libsolidity/codegen/CompilerContext.cpp [hunk 2]
```diff
@@ -124,14 +124,15 @@ void CompilerContext::addVariable(VariableDeclaration const& _declaration,
...
 }
 
 void CompilerContext::removeVariable(VariableDeclaration const& _declaration)
 {
-	solAssert(!!m_localVariables.count(&_declaration), "");
-	m_localVariables.erase(&_declaration);
+	solAssert(m_localVariables.count(&_declaration) && !m_localVariables[&_declaration].empty(), "");
+	m_localVariables[&_declaration].pop_back();
+	if (m_localVariables[&_declaration].empty())
+		m_localVariables.erase(&_declaration);
```
Comment (axic) [inline libsolidity/codegen/CompilerContext.cpp:135]: As far as I see this is just a memory conservation piece, it has no effect.

**Review by axic — APPROVED** [review]

File: libsolidity/codegen/CompilerContext.cpp (hunk 2 above)
Comment (chriseth) [reply libsolidity/codegen/CompilerContext.cpp:135]: I do not avoid the ambiguous situation where the declaration is present in the map as a key but is not actually a current local variable.

**Pull request merged**